src/plugins/intel_cpu/tests/functional/subgraph_tests/src/ngram.cpp

```diff
@@ -0,0 +1,232 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <tuple>
+#include <string>
+#include <vector>
+#include <memory>
+#include <debug.h>
+#include <shared_test_classes/base/ov_subgraph.hpp>
+#include <ngraph_functions/builders.hpp>
+#include "common_test_utils/common_utils.hpp"
+#include <common_test_utils/ov_tensor_utils.hpp>
+#include "test_utils/cpu_test_utils.hpp"
+#include <openvino/opsets/opset1.hpp>
+
+using namespace CPUTestUtils;
+using namespace ov::test;
+using namespace ngraph::helpers;
+
+namespace CPUSubgraphTestsDefinitions {
+
+typedef std::tuple<
+    std::vector<InputShape>,
+    ElementType,
+    ElementType,
+    size_t
+> NgramTestParams;
+
+static std::shared_ptr<ov::Node> getStridedSlice(const std::shared_ptr<ov::Node> data,
```

Comment:
Please use references for shared_ptr as well, due to their copying produce additional routine.